### ISSUE TITLE
Issue with using mouse scroll wheel to zoom body pane, with custom styling

### ISSUE THREAD

**Issue by devadrian1** [leo-editor/leo-editor#490]
I am finding that if I edit the syntax styling of a language element (e.g. "keyword1" for PHP), the element takes on the new styling, but the **re-sizing** using the mouse scroll wheel no longer works, **just** for that element.

For example, let's say that we want to make the "keyword1" group for **@language php** bold, I can put this in my outline, or _myLeoSettings_, etc and then use **Settings->Reload Styles** from the top menu.

```
@settings
      @font php
             (in body pane) keyword1_font_weight=bold
```
I noticed this same behavior with **@language php** and **@language python** also.

**EKR**: I have put the images in another (hidden) comment.

**Comment by edreamleo** [comment]
I marked this as lvl:minor because the workaround is to reload Leo rather than using the reload styles command.

**Comment by tbnorth** [comment]
Not sure reload-styles / reloading Leo impacts the bug?  @devadrian1?
Don't think Ctrl-mousewheel resizing uses reload-styles?

**Comment by devadrian1** [comment]
Hi,

Firstly, sorry for not submitting my PHP pack.....been too busy with work
(even after hours) to make it professional quality.


Regarding the zoom problem....I just have no idea really......I make some
changes....it works..then stops working...I know there is some reason...I
just do not understand Qt enough to know why.  It currently is working for
me on Courier New font.

Suggestion:

It is somewhat confusing to tell us we can edit a stylesheet AND also
adjust by Leo attributes.  Might want to think about that.

Thanks,
Adrian

**Comment by vitalije** [comment]
>I marked this as lvl:minor because the workaround is to reload Leo rather than using the reload styles command.

@edreamleo I am not sure you understand what this issue is about. It is not about reloading styles or Leo. Even if user reopens Leo, Ctrl + mouse scroll won't work for any part of the text that has special styling defined in the settings tree. Look at the picture above, the word `function` is not enlarged when using Ctrl+mouse scroll, while the rest of the text is. 

Here is the example file. It is gzipped because github refused ordinary Leo file attachment.
[font-issue.leo.gz](https://github.com/leo-editor/leo-editor/files/2790355/font-issue.leo.gz)

Open it and try zooming in and out with Ctrl+mouse scroll. Python keywords like: while, if, else, break, ... are not resized.

**Comment by edreamleo** [comment]
@vitalije I agree: this issue is not solved by reloading Leo.

**Comment by vitalije** [comment]
Perhaps one solution would be to remove support for at-font settings related to syntax colorizing. Other uses of at-font can continue as they are now. Also documentation should be adjusted and leoSettings.leo. Most users will never notice this bug if they use and tweak only themes. However if one tries to add setting `@font <language>_keyword<1,2,3..>_font` then bug is evident. Official leoSettings.leo contains only font definitions for languages like forth and php. Those settings should be removed also. This explains why not so many users have noticed this bug.

I remember that I was looking into this issue before, but without any success. IIRC the problem is caused with two separate ways of styling body text. `@font` was earlier in the pre - Qt days. With Qt came qt-style-sheets but, the old code was not removed and now it may be challenging to make both schemes work together well. After all Ctrl+mouse scrolling was not available before Leo switched to Qt.

**Comment by vitalije** [comment]
@edreamleo,  have you read my last comment in this thread? The label **won't fix** and closing this issue has almost the same effect, but the proposed fix would be cleaner. The fact that so few users had noticed this issue is because they don't use `@font <lang>_*` settings. In leoSettings.leo there is just one such setting for PHP keywords and a few settings for Forth programming language. The offended party of this issue mentions PHP. Fix would be just to remove `@font php keyword fonts` node from leoSettings.leo. But clean fix would be to remove also settings for Forth and to remove parts of the documentation mentioning font settings per language.

**Comment by edreamleo** [comment]
@vitalije I don't remember your comment, but that's a statement only about my memory ;-)

I'll reopen this, provided you are willing to work on it.  I'm concerned about removing settings.  How do you propose to allow per-language settings?

**Comment by vitalije** [comment]
The fix I proposed was to remove support for font per language settings. Obviously if one uses it zoom with mouse wheel won't work as expected. Removing mentions of this feature in the documentation and removing those settings from the official leoSettings.leo will solve the issue.

**Comment by edreamleo** [comment]
Here are the original images:

See image .1 
![leo-issue-1](https://cloud.githubusercontent.com/assets/22187576/26106726/68e8800e-3a0c-11e7-93e5-081143b9f92c.png)


Image .2 contains the PHP source of some test gobbly guk code _before_ attempting to change text size with scroll wheel.
![leo-issue-2](https://cloud.githubusercontent.com/assets/22187576/26106746/7e949460-3a0c-11e7-98bd-1af34abca474.png)


Image .3 contains the PHP source _after_ scrolling the text size up a few motions.
![leo-issue-3](https://cloud.githubusercontent.com/assets/22187576/26106748/80e7dc86-3a0c-11e7-9d8e-f2b6d10a9a1b.png)

**Comment by edreamleo** [comment]
The fix is in devel at rev 537af3b.
